sprocs/instance_questions_check_sequence_locked.sql

```diff
@@ -0,0 +1,61 @@
+-- Given an instance question id `iq_id`, returns a boolean that's true if an
+-- assessment's 'minAdvancePerc' settings should prevent user access.
+CREATE OR REPLACE FUNCTION
+    instance_questions_check_sequence_locked (
+        iq_id bigint
+    ) RETURNS boolean AS $$
+DECLARE
+    ai_id bigint;
+    prev_iq_open boolean;
+    prev_aq_id bigint;
+    prev_iq_highest_submission_score double precision;
+    relevant_min_advance_perc double precision;
+    prev_iq_score_met boolean;
+
+BEGIN
+
+    SELECT iq.assessment_instance_id, iq.open INTO ai_id
+    FROM instance_questions iq
+    WHERE iq.id = iq_id;
+
+    -- 1. Get the id and score_perc of the previous instance question in the assessment.
+    SELECT _prev_aq_id, _prev_iq_open, _prev_iq_highest_submission_score INTO prev_aq_id, prev_iq_open, prev_iq_highest_submission_score
+    FROM (
+        SELECT
+            iq.id AS cur_iq,
+            (lag(aq.id) OVER w) AS _prev_aq_id,
+            (lag(iq.open) OVER w) AS _prev_iq_open,
+            (lag(iq.highest_submission_score) OVER w) AS _prev_iq_highest_submission_score
+        FROM
+            assessment_instances AS ai
+            JOIN assessments AS a ON (a.id = ai.assessment_id)
+            JOIN instance_questions AS iq ON (iq.assessment_instance_id = ai.id)
+            JOIN question_order(ai.id) AS qo ON (qo.instance_question_id = iq.id)
+            JOIN assessment_questions AS aq ON (aq.id = iq.assessment_question_id)
+        WHERE
+            ai.id IN (SELECT iq.assessment_instance_id FROM instance_questions iq WHERE iq.id = iq_id)
+        WINDOW
+            w AS (ORDER BY qo.row_order)
+    ) AS prev_iq
+    WHERE cur_iq = iq_id;
```

Comment:
If I'm understanding correctly, this only blocks progress to the very next question. I guess I would have assumed that if a question has a `minAdvancePerc` set, then you wouldn't be able to progress to any question past it. Otherwise, how do we do things like say "you have to do the first 5 questions in order, but then the next 5 you can do anyway you want"?